commit/factory.go

```diff
@@ -159,7 +159,7 @@ func (p *PluginFactory) NewReportingPlugin(ctx context.Context, config ocr3types
 	)
 
 	// The node supports the chain that the token prices are on.
-	tokenPricesCr, ok := p.contractReaders[offchainConfig.PriceFeedChainSelector]
+	_, ok := readers[offchainConfig.PriceFeedChainSelector]
```

Comment:
I'm surprised that you need to switch from `p.contractReaders` to `readers`. Is that really required?

We can move the facade mapping to `NewPluginFactory` and avoid using the actual ContractReader type outside of the constructor.

Reply:
I don't "need" to, it felt more consistent to use the same `facade` everywhere though.